File: src/main/java/de/tectoast/emolga/commands/music/control/NpCommand.java
package de.tectoast.emolga.commands.music.control;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.commands.MusicCommand;
import de.tectoast.emolga.utils.music.GuildMusicManager;
import net.dv8tion.jda.api.entities.Member;
import net.dv8tion.jda.api.entities.Message;
import net.dv8tion.jda.api.entities.TextChannel;

public class NpCommand extends MusicCommand {
    public NpCommand() {
        super("np", "Zeigt, welcher Track gerade läuft");
        setArgumentTemplate(ArgumentManagerTemplate.noArgs());
    }

    @Override
    public void process(GuildCommandEvent e) {
        TextChannel tco = e.getChannel();
        Message m = e.getMessage();
        String msg = m.getContentDisplay();
        Member member = e.getMember();
        GuildMusicManager manager = getGuildAudioPlayer(tco.getGuild());
        if (manager.player == null) {
            tco.sendMessage("Derzeit läuft kein Track!").queue();
            return;
        }
        AudioPlayer player = manager.player;
        if (player.getPlayingTrack() == null) {
            tco.sendMessage("Derzeit läuft kein Track!").queue();
            return;
        }
        AudioTrack t = player.getPlayingTrack();
        long p = t.getPosition();
        int cm = (int) (p / 60000);
        int cs = (int) ((p - cm * 60000) / 1000);
        long d = t.getDuration();
        int dm = (int) (d / 60000);
        int ds = (int) ((d - dm * 60000) / 1000);
        tco.sendMessage("`" + t.getInfo().title + "`\n" + getWithZeros(cm, 2) + ":" + getWithZeros(cs, 2) + " / " + getWithZeros(dm, 2) + ":" + getWithZeros(ds, 2)).queue();
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/music/control/NpCommand.kt
package de.tectoast.emolga.commands.music.control

import de.tectoast.emolga.commands.GuildCommandEvent
import de.tectoast.emolga.commands.MusicCommand

class NpCommand : MusicCommand("np", "Zeigt, welcher Track gerade läuft") {
    init {
        argumentTemplate = ArgumentManagerTemplate.noArgs()
    }

    override fun process(e: GuildCommandEvent) {
        val tco = e.textChannel
        val manager = getGuildAudioPlayer(tco.guild)
        val player = manager.player
        if (player.playingTrack == null) {
            tco.sendMessage("Derzeit läuft kein Track!").queue()
            return
        }
        val t = player.playingTrack
        val p = t.position
        val cm = (p / 60000).toInt()
        val cs = ((p - cm * 60000) / 1000).toInt()
        val d = t.duration
        val dm = (d / 60000).toInt()
        val ds = ((d - dm * 60000) / 1000).toInt()
        tco.sendMessage(
            "`${t.info.title}`\n${getWithZeros(cm, 2)}:${getWithZeros(cs, 2)} / ${
                getWithZeros(dm, 2)
            }:${getWithZeros(ds, 2)}"
        ).queue()
    }
}